src/Pages/MapPage.jsx

```diff
@@ -46,48 +48,169 @@ class MapPage extends Component {
   buttonList(props) {
     
     //Store listButtons:
-    const hallNames = props.hallNames;
-    const hallLocations = props.hallCoordinates;
-
-    //zip the two arrays together:
-    var hallNamesAndLocations = hallNames.map((x, i) => [x, hallLocations[i]]); 
-
-    console.log(hallNamesAndLocations);
-    console.log(hallNames.length);
+    const hallKV = props.hallKVPair;
 
     //Gemerate hall buttons.
-    const listButtonItems = hallNamesAndLocations.map((hallInfo) =>
+    const listButtonItems = hallKV.map((hallInfo) =>
       <Button
         variant="success"
         size="lg"
         className={MapPageStyles.hallButton}
-        onMouseOver={() => { this.changeCenterHover(hallInfo[1]) }}
-        onClick={() => { this.navigateToPage(hallInfo[0].toString()) }}
-      >{hallInfo[0]}
+        onMouseOver={() => { this.changeCenterHover(hallInfo.value) }}
+        onClick={() => { this.navigateToPage(hallInfo.key.toString()) }}
+      >{hallInfo.key}
       </Button>
     );
 
     //Generate markers for buttons.
-    this.markers = hallNamesAndLocations.map((hallInfo) =>
+    this.markers = hallKV.map((hallInfo) =>
       <Marker 
-        position={hallInfo[1]}
+        position={hallInfo.value}
       >
         <Popup>
-          {hallInfo[0]}
+          {hallInfo.key}
         </Popup>
       </Marker>
     );
     
     return(
-      <div 
-        className={MapPageStyles.listColumn}
-        onMouseLeave={this.reCenter}
-      >
-        {listButtonItems}
+      <div className={MapPageStyles.listColumn} onMouseLeave={this.reCenter}>
+
+        <div className={MapPageStyles.optionsBlock}>
+          <this.hallOptions/>
+        </div>
+
+        <div className={MapPageStyles.buttonListBlock}>
+          {listButtonItems}
+        </div>
+
       </div>
     );
   }
 
+  //===hallOptions===
+  //Desc: Get hall option inputs.
+  hallOptions() {
+
+    return (
+     <Form>
+        <Form.Group>
+          <h5>Filter:</h5>
+          <Form.Control as="select" defaultValue="Choose..." onChange={(e)=>{this.mapChangeOptions(e.target.value)}}>
+            <option>Choose</option>
+            <option>Alphabetical</option>
+            <option>Residents Halls</option>
+            <option>Apartments</option>
+            <option>Bassettis</option>
+            <option>Central</option>
+            <option>North</option>
+            <option>South</option>
+          </Form.Control>
+          <Button variant="primary" onClick={()=>{this.mapChangeOptions("Reset")}} className={MapPageStyles.resetButton}>
+            Reset
+          </Button>
+        </Form.Group>
+      </Form>
+    );
+
+  }
+
+  //===mapChangeOptions===
+  //Desc: function to chnage the map with the filers.
+  mapChangeOptions(option) {
+    
+    if(option == "Alphabetical")
```

Comment:
All double equals should be triple equals (===)
This prevents against annoying js bugs and fixes warnings we get